### PR TITLE
[DATA-6469] Make plan, feature link, table link as dropdown button

### PR DESCRIPTION
**Pull request by youngyjd**
### Summary of Changes

See detailed commit message in https://github.com/brexhq/credit_card/pull/78055 as this repo is public

### CheckList
Make sure you have checked **all** steps below to ensure a timely review.
- [ ] PR title addresses the issue accurately and concisely. Example: "Updates the version of Flask to v1.0.2"
    - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
- [ ] PR includes a summary of changes.
- [ ] PR adds unit tests, updates existing unit tests, __OR__ documents why no test additions or modifications are needed.
- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
    - All the public functions and the classes in the PR contain docstrings that explain what it does

### CODE REVIEW

File: frontend/amundsen_application/static/css/_layouts.scss [hunk 1]
```diff
@@ -26,6 +26,8 @@ $icon-header-size: 32px;
     }
 
     .header-section {
+      flex-shrink: 0;
```
Comment (youngyjd) [inline frontend/amundsen_application/static/css/_layouts.scss:29]: context for the change of this line: https://github.com/amundsen-io/amundsen/pull/1855

it has been merged into upstream.

File: frontend/amundsen_application/static/js/pages/TableDetailPage/ApplicationDropdown/index.tsx [hunk 2]
```diff
@@ -32,6 +31,14 @@ const getImagePath = (tableAppName) => {
...
     case DATABRICKS.toLowerCase():
       return DATABRICKS_LOGO_PATH;
+    case 'feature_link':
+      return '/static/images/icons/Database.svg';
+    case 'snowflake_link':
+      return '/static/images/icons/logo-snowflake.svg';
+    case 'postgres_link':
+      return '/static/images/icons/logo-postgres.svg';
+    case 'plan':
+      return '/static/images/github.png';
```
Comment (youngyjd) [inline frontend/amundsen_application/static/js/pages/TableDetailPage/ApplicationDropdown/index.tsx:41]: This determines the logo for the dropdown button

File: frontend/amundsen_application/static/js/pages/TableDetailPage/ApplicationDropdown/index.tsx [hunk 3]
```diff
@@ -123,7 +130,11 @@ const ApplicationDropdown: React.FC<ApplicationDropdownProps> = ({
   const image = getImagePath(tableApps[0].name.toLowerCase());
   const avatarLabel = isAirflowOrDatabricksApp(tableApps[0].name)
     ? tableApps[0].name
-    : APPLICATIONS_LABEL;
+    : (
+        tableApps[0].name.charAt(0).toUpperCase() + tableApps[0].name.slice(1)
+      ).replace('_link', '');
```
Comment (youngyjd) [inline frontend/amundsen_application/static/js/pages/TableDetailPage/ApplicationDropdown/index.tsx:135]: this determines the dropdown button text

**Review by wli-brex — APPROVED** [review]
💯

**Pull request merged**